lib/rubocop/cop/rspec/stubbed_mock.rb

```diff
@@ -0,0 +1,172 @@
+        #   @param node [RuboCop::AST::Node]
+        #   @yield [RuboCop::AST::Node] expectation, method name, matcher
+        def_node_matcher :expectation, <<~PATTERN
+          (send
+            $(send nil? $#{Expectations::ALL.node_pattern_union} ...)
+            :to $_)
+        PATTERN
+
+        # @!method matcher_with_configured_response(node)
+        #   Match matcher with a configured response
+        #
+        #   @example source that matches
+        #     receive(:foo).and_return('bar')
+        #
+        #   @example source that matches
+        #     receive(:lower).and_raise(SomeError)
+        #
+        #   @example source that matches
+        #     receive(:redirect).and_call_original
+        #
+        #   @param node [RuboCop::AST::Node]
+        #   @yield [RuboCop::AST::Node] matcher
+        def_node_matcher :matcher_with_configured_response, <<~PATTERN
+          (send #message_expectation? #configured_response? _)
+        PATTERN
+
+        # @!method matcher_with_return_block(node)
+        #   Match matcher with a return block
+        #
+        #   @example source that matches
+        #     receive(:foo) { 'bar' }
+        #
+        #   @param node [RuboCop::AST::Node]
+        #   @yield [RuboCop::AST::Node] matcher
+        def_node_matcher :matcher_with_return_block, <<~PATTERN
+          (block #message_expectation? args _)  # receive(:foo) { 'bar' }
+        PATTERN
+
+        # @!method matcher_with_hash(node)
+        #   Match matcher with a configured response defined as a hash
+        #
+        #   @example source that matches
+        #     receive_messages(foo: 'bar', baz: 'qux')
+        #
+        #   @example source that matches
+        #     receive_message_chain(:foo, bar: 'baz')
+        #
+        #   @param node [RuboCop::AST::Node]
+        #   @yield [RuboCop::AST::Node] matcher
+        def_node_matcher :matcher_with_hash, <<~PATTERN
+          {
+            (send nil? :receive_messages hash)           # receive_messages(foo: 'bar', baz: 'qux')
+            (send nil? :receive_message_chain ... hash)  # receive_message_chain(:foo, bar: 'baz')
+          }
+        PATTERN
+
+        # @!method matcher_with_blockpass(node)
+        #   Match matcher with a configured response in block-pass
+        #
+        #   @example source that matches
+        #     receive(:foo, &canned)
+        #
+        #   @example source that matches
+        #     receive_message_chain(:foo, :bar, &canned)
+        #
+        #   @example source that matches
+        #     receive(:foo).with('bar', &canned)
+        #
+        #   @param node [RuboCop::AST::Node]
+        #   @yield [RuboCop::AST::Node] matcher
+        def_node_matcher :matcher_with_blockpass, <<~PATTERN
+          {
+            (send nil? { :receive :receive_message_chain } ... block_pass)  # receive(:foo, &canned)
+            (send (send nil? :receive ...) :with ... block_pass)            # receive(:foo).with('foo', &canned)
```

Comment:
Wondering if you guys find that kind of node pattern comments helpful @bquorning @Darhazer

Reply:
I found them very useful, thank you @pirj